cmd/cluster/context_test.go

```diff
@@ -0,0 +1,829 @@
+package cluster
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"bou.ke/monkey"
```

Comment:
Can we just write interfaces where you need them instead of pulling in yet another mocking library/framework?

Reply:
Thanks for the feedback. I used the mocking framework as it required minimal changes. Since there are only a few interfaces, I’ll need to create additional ones to mock the behavior, which will involve more code changes. I’ll proceed with that approach as per your suggestion.

Reply:
Implemented the unit test cases using interfaces instead of mocking library as per the suggestion.